<a href="https://colab.research.google.com/github/juanjose1919/Proyectos-segundo-corte/blob/main/PROYECTO_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

$$\huge\Proyecto\5: $$ 

#Introduccion
Dentro de este proyecto se analizará la base de datos referida a los clientes del mercado financiero, mediante los préstamos que estos adquieren, este nivel de préstamos depende de variables como ingreso, edad, el nivel educativo, a través del proyecto se compararán las variables a través de pruebas de proporciones, diferencias de medias y medianas, analizando también como se distribuyen los datos por medio de la prueba de normalidad y realizando finalmente tablas de contingencia que nos permitan visualizar mas eficientemente las relaciones proporcionales entre las variables


In [ ]:
lista = [['- -', '- -','- -','CONTENIDO', '- -',' - -', '- -'], ['1.', 'Exploración inicial de los datos'], ['2.', 'Limpieza de los datos'],
         ['3.', 'Diferencia de proporciones'],
         ['4.', 'Distribución norma'],
         ['5.', 'Media o en la mediana significativas'],
         ['6.', 'Tablas de contingencia'],
         ['6.', 'Prueba X^2'],
         ['7.', 'Conclusión general']]

In [ ]:
for fila in lista:
  for elem in fila:
    print(f'{elem:<5}', end='')
  print()

- -  - -  - -  CONTENIDO- -   - - - -  
1.   Exploración inicial de los datos
2.   Limpieza de los datos
3.   Diferencia de proporciones
4.   Distribución norma
5.   Media o en la mediana significativas
6.   Tablas de contingencia
6.   Prueba X^2
7.   Conclusión general


In [ ]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt 
import scipy.stats as st 
from scipy.stats import stats
data = pd.read_csv('https://raw.githubusercontent.com/labeconometria/MLxE/main/proyectos1er/dataset_2.csv')
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from statsmodels.stats.weightstats import ztest as ztest

#1 Realice una exploración inicial de los datos

In [ ]:
data

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
0         High School        Married     $60K - $80K          Blue   
1            Graduate         Single  Less than $40K          Blue   
2            Graduate        Married    $80K - $120K          Blue   
3         High School        Unknown  Less than $40K          Blue   
4          Uneducated        Married     $60K - $80K          Blue   
...               ...            ...             ...           ...   
10122        Graduate         Single     $40K - $60K          Blue   
10123         Unknown       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  \
0                  39  ...       12691.0                  777   
1                  44  ...        8256.0                  864   
2                  36  ...        3418.0                    0   
3                  34  ...        3313.0                 2517   
4                  21  ...        4716.0                    0   
...               ...  ...           ...                  ...   
10122              40  ...        4003.0                 1851   
10123              25  ...        4277.0                 2186   
10124              36  ...        5409.0                    0   
10125              36  ...        5281.0                    0   
10126              25  ...       10388.0                 1961   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0              11914.0                 1.335             1144              42   
1               7392.0                 1.541             1291              33   
2               3418.0                 2.594             1887              20   
3                796.0                 1.405             1171              20   
4               4716.0                 2.175              816              28   
...                ...                   ...              ...             ...   
10122           2152.0                 0.703            15476             117   
10123           2091.0                 0.804             8764              69   
10124           5409.0                 0.819            10291              60   
10125           5281.0                 0.535             8395              62   
10126           8427.0                 0.703            10294              61   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
0                    1.625                  0.061   
1                    3.714                  0.105   
2                    2.333                  0.000   
3                    2.333                  0.760   
4                    2.500                  0.000   
...                    ...                    ...   
10122                0.857                  0.462   
10123                0.683                  0.511   
10124     

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10127 entries, 0 to 10126
Data columns (total 23 columns):
 #   Column                                                                                                                              Non-Null Count  Dtype  
---  ------                                                                                                                              --------------  -----  
 0   CLIENTNUM                                                                                                                           10127 non-null  int64  
 1   Attrition_Flag                                                                                                                      10127 non-null  object 
 2   Customer_Age                                                                                                                        10127 non-null  int64  
 3   Gender                                                                           

In [ ]:
data.head()

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0  768805383  Existing Customer            45      M                3   
1  818770008  Existing Customer            49      F                5   
2  713982108  Existing Customer            51      M                3   
3  769911858  Existing Customer            40      F                4   
4  709106358  Existing Customer            40      M                3   

  Education_Level Marital_Status Income_Category Card_Category  \
0     High School        Married     $60K - $80K          Blue   
1        Graduate         Single  Less than $40K          Blue   
2        Graduate        Married    $80K - $120K          Blue   
3     High School        Unknown  Less than $40K          Blue   
4      Uneducated        Married     $60K - $80K          Blue   

   Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  \
0              39  ...       12691.0                  777          11914.0   
1              44  ...        8256.0                  864           7392.0   
2              36  ...        3418.0                    0           3418.0   
3              34  ...        3313.0                 2517            796.0   
4              21  ...        4716.0                    0           4716.0   

   Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  Total_Ct_Chng_Q4_Q1  \
0                 1.335             1144              42                1.625   
1                 1.541             1291              33                3.714   
2                 2.594             1887              20                2.333   
3                 1.405             1171              20                2.333   
4                 2.175              816              28                2.500   

   Avg_Utilization_Ratio  \
0                  0.061   
1                  0.105   
2                  0.000   
3                  0.760   
4                  0.000   

   Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1  \
0                                           0.000093                                                                                    
1                                           0.000057                                                                                    
2                                           0.000021                                                                                    
3                                           0.000134                                                                                    
4                                           0.000022                                                                                    

   Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2  
0                                            0.99991                                                                                   
1                                            0.99994                                                                                   
2                                            0.99998                                                                                   
3                                            0.99987                                                                                   
4                                            0.99998                                                                                   

[5 rows x 23 columns]

In [ ]:
data.describe()

CLIENTNUM  Customer_Age  Dependent_count  Months_on_book  \
count  1.012700e+04  10127.000000     10127.000000    10127.000000   
mean   7.391776e+08     46.325960         2.346203       35.928409   
std    3.690378e+07      8.016814         1.298908        7.986416   
min    7.080821e+08     26.000000         0.000000       13.000000   
25%    7.130368e+08     41.000000         1.000000       31.000000   
50%    7.179264e+08     46.000000         2.000000       36.000000   
75%    7.731435e+08     52.000000         3.000000       40.000000   
max    8.283431e+08     73.000000         5.000000       56.000000   

       Total_Relationship_Count  Months_Inactive_12_mon  \
count              10127.000000            10127.000000   
mean                   3.812580                2.341167   
std                    1.554408                1.010622   
min                    1.000000                0.000000   
25%                    3.000000                2.000000   
50%                    4.000000                2.000000   
75%                    5.000000                3.000000   
max                    6.000000                6.000000   

       Contacts_Count_12_mon  Credit_Limit  Total_Revolving_Bal  \
count           10127.000000  10127.000000         10127.000000   
mean                2.455317   8631.953698          1162.814061   
std                 1.106225   9088.776650           814.987335   
min                 0.000000   1438.300000             0.000000   
25%                 2.000000   2555.000000           359.000000   
50%                 2.000000   4549.000000          1276.000000   
75%                 3.000000  11067.500000          1784.000000   
max                 6.000000  34516.000000          2517.000000   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
count     10127.000000          10127.000000     10127.000000    10127.000000   
mean       7469.139637              0.759941      4404.086304       64.858695   
std        9090.685324              0.219207      3397.129254       23.472570   
min           3.000000              0.000000       510.000000       10.000000   
25%        1324.500000              0.631000      2155.500000       45.000000   
50%        3474.000000              0.736000      3899.000000       67.000000   
75%        9859.000000              0.859000      4741.000000       81.000000   
max       34516.000000              3.397000     18484.000000      139.000000   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
count         10127.000000           10127.000000   
mean              0.712222               0.274894   
std               0.238086               0.275691   
min               0.000000               0.000000   
25%               0.582000               0.023000   
50%               0.702000               0.176000   
75%               0.818000               0.503000   
max               3.714000               0.999000   

       Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1  \
count                                       10127.000000                                                                                    
mean                                            0.159997                                                                                    
std                                             0.365301                                                                                    
min                                             0.000008                                                                                    
25%                                             0.000099                                                                                    
50%                                             0.000181                                                                                    
75%                                             

In [ ]:
data.tail()

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
10122        Graduate         Single     $40K - $60K          Blue   
10123         Unknown       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  \
10122              40  ...        4003.0                 1851   
10123              25  ...        4277.0                 2186   
10124              36  ...        5409.0                    0   
10125              36  ...        5281.0                    0   
10126              25  ...       10388.0                 1961   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
10122           2152.0                 0.703            15476             117   
10123           2091.0                 0.804             8764              69   
10124           5409.0                 0.819            10291              60   
10125           5281.0                 0.535             8395              62   
10126           8427.0                 0.703            10294              61   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
10122                0.857                  0.462   
10123                0.683                  0.511   
10124                0.818                  0.000   
10125                0.722                  0.000   
10126                0.649                  0.189   

       Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1  \
10122                                           0.000191                                                                                    
10123                                           0.995270                                                                                    
10124                                           0.997880                                                                                    
10125                                           0.996710                                                                                    
10126                                           0.996620                                                                                    

       Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2  
10122                                           0.999810                                                                                   
10123                                           0.004729                                                                                   
10124                                           0.002118                                                                                   
10125                                           0.003294                                                                                   
10126                                           0.003377                                                                                   

[5 rows x 23 columns]

#2 Realice una limpieza de los datos

In [ ]:
data.duplicated().sum()

0

In [ ]:
data.isna().sum()

CLIENTNUM                                                                                                                             0
Attrition_Flag                                                                                                                        0
Customer_Age                                                                                                                          0
Gender                                                                                                                                0
Dependent_count                                                                                                                       0
Education_Level                                                                                                                       0
Marital_Status                                                                                                                        0
Income_Category                                 

Conclusión intermedia: se encuentra que no existen datos duplicados ni missing values, teniendo en cuenta esto, no es necesario llevar a cabo limpieza de los datos

In [ ]:
data.nunique()

CLIENTNUM                                                                                                                             10127
Attrition_Flag                                                                                                                            2
Customer_Age                                                                                                                             45
Gender                                                                                                                                    2
Dependent_count                                                                                                                           6
Education_Level                                                                                                                           7
Marital_Status                                                                                                                            4
Income_Category     

In [ ]:
datas=data['CLIENTNUM']
q1=datas.quantile(0.25)
q3=datas.quantile(0.75)
rq=q3-q1
lims=(1.5*rq)+q3 
w=datas[(datas>lims)].index
w

Int64Index([], dtype='int64')

#3 Verifique si la diferencia de proporciones para las siguientes variables es estadísticamente significativa:

In [ ]:
data1 = data[['Attrition_Flag','Gender']]
data1.replace({'Gender':{'F':1, 'M' :0}}, inplace = True)
data1.replace({'Attrition_Flag':{'Existing Customer':1, 'Attrited Customer' :0}}, inplace = True)

/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1951: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[selected_item_labels] = value
/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1773: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(ilocs[0], value, pi)


In [ ]:
data1

Attrition_Flag  Gender
0                   1       0
1                   1       1
2                   1       0
3                   1       1
4                   1       0
...               ...     ...
10122               1       0
10123               0       0
10124               0       1
10125               0       0
10126               0       1

[10127 rows x 2 columns]

In [ ]:
pd.crosstab(data1['Gender'], data1['Attrition_Flag'])/len(data) *100

Attrition_Flag         0          1
Gender                             
0               6.882591  40.209341
1               9.183371  43.724696

In [ ]:
n = len(data['Gender'])
x1 = data1['Attrition_Flag']
x2 = data1['Gender']


In [ ]:
z_test ,propability_value =  ztest(x1, x2, alternative="larger")
print(float(propability_value))
if propability_value<0.05:
 print("Se rechaza la hipótesis nula , hay diferencia significativa de proporciones")
else: 
 print("Se acepta la hipótesis nula , no hay diferencia significativa de proporciones")

0.0
Se rechaza la hipótesis nula , hay diferencia significativa de proporciones


**Conclusión intermedia:** A partir de la prueba z para determinar diferencia significativa entre las proporciones de las variables Attrition_Flag y Gender se encuentra que si existe diferencia significativa entre las proporciones de dichas variables

In [ ]:
ztest(x1, value= n)

(-2774861.719756017, 0.0)

# 4 Verifique si las siguientes variables siguen una distribución normal

In [ ]:
data

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
0         High School        Married     $60K - $80K          Blue   
1            Graduate         Single  Less than $40K          Blue   
2            Graduate        Married    $80K - $120K          Blue   
3         High School        Unknown  Less than $40K          Blue   
4          Uneducated        Married     $60K - $80K          Blue   
...               ...            ...             ...           ...   
10122        Graduate         Single     $40K - $60K          Blue   
10123         Unknown       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Credit_Limit  Total_Revolving_Bal  \
0                  39  ...       12691.0                  777   
1                  44  ...        8256.0                  864   
2                  36  ...        3418.0                    0   
3                  34  ...        3313.0                 2517   
4                  21  ...        4716.0                    0   
...               ...  ...           ...                  ...   
10122              40  ...        4003.0                 1851   
10123              25  ...        4277.0                 2186   
10124              36  ...        5409.0                    0   
10125              36  ...        5281.0                    0   
10126              25  ...       10388.0                 1961   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0              11914.0                 1.335             1144              42   
1               7392.0                 1.541             1291              33   
2               3418.0                 2.594             1887              20   
3                796.0                 1.405             1171              20   
4               4716.0                 2.175              816              28   
...                ...                   ...              ...             ...   
10122           2152.0                 0.703            15476             117   
10123           2091.0                 0.804             8764              69   
10124           5409.0                 0.819            10291              60   
10125           5281.0                 0.535             8395              62   
10126           8427.0                 0.703            10294              61   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  \
0                    1.625                  0.061   
1                    3.714                  0.105   
2                    2.333                  0.000   
3                    2.333                  0.760   
4                    2.500                  0.000   
...                    ...                    ...   
10122                0.857                  0.462   
10123                0.683                  0.511   
10124     

In [ ]:
data_2 = data[['Customer_Age', 'Dependent_count']]

In [ ]:
data_2

Customer_Age  Dependent_count
0                45                3
1                49                5
2                51                3
3                40                4
4                40                3
...             ...              ...
10122            50                2
10123            41                2
10124            44                1
10125            30                2
10126            43                2

[10127 rows x 2 columns]

In [ ]:
n= len(data_2['Customer_Age'])
n

10127

In [ ]:
p_valor_customer = st.shapiro(data_2['Customer_Age'])
p_valor_customer

/usr/local/lib/python3.7/dist-packages/scipy/stats/morestats.py:1760: UserWarning: p-value may not be accurate for N > 5000.
  warnings.warn("p-value may not be accurate for N > 5000.")


ShapiroResult(statistic=0.9960706233978271, pvalue=6.600162251926681e-16)

**Conclusión intermedia**: 

Ho(Aceptado): La muestra es de las distribuciones normales. (Po>0.05)

Ha(Rechazado): La muestra no es de las distribuciones normales.

Analizando la variable Customer_Age mediante la prueba de normalidad de Shapiro-Wilk, se encuentra un p-valor de 6.600162251926681e-16 el cual es menor a 0.05 por lo que se rechaza la hipótesis nula, es decir, existe suficiente evidencia estadística para afirmar que la muestra no proviene de una distribución normal.

In [ ]:
p_valor_dependent = st.shapiro(data_2['Dependent_count'])
p_valor_dependent

ShapiroResult(statistic=0.936938464641571, pvalue=0.0)

**Conclusión intermedia**: 

Ho(Aceptado): La muestra es de las distribuciones normales. (Po>0.05)

Ha(Rechazado): La muestra no es de las distribuciones normales.

Analizando la variable Dependent_count mediante la prueba de normalidad de Shapiro-Wilk, se encuentra un p-valor de 0.0 el cual es menor a 0.05 por lo que se rechaza la hipótesis nula, es decir, existe suficiente evidencia estadística para afirmar que la muestra no proviene de una distribución normal.

#5 Divida el dataset en dos grupos a partir de la variable Attrition Flag. Verifique si hay diferencias en la media o en la mediana significativas en cuanto a Customer Age y en cuanto a Dependent Count para los dos grupos.

In [ ]:
data3 = data[['Attrition_Flag','Customer_Age', 'Dependent_count']]
data3.replace({'Attrition_Flag':{'Existing Customer':1, 'Attrited Customer' :0}}, inplace = True)
Existing_Customer = []
Attrited_Customer = []
for i in  data3['Attrition_Flag']:
  Existing_Customer.append(i)
  if i == i:
    if i == 1:
      i=0
      Attrited_Customer.append(i)
    else:
      i=1
      Attrited_Customer.append(i)
      

data3['Existing_Customer'] = Existing_Customer
data3['Attrited_Customer'] = Attrited_Customer

/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1773: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(ilocs[0], value, pi)
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  app.launch_new_instance()
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instea

In [ ]:
#data3

In [ ]:
data3.groupby('Existing_Customer').agg({'Customer_Age':['mean','median'], 'Dependent_count':['mean','median']})

Customer_Age        Dependent_count       
                          mean median            mean median
Existing_Customer                                           
0                    46.659496   47.0        2.402581    2.0
1                    46.262118   46.0        2.335412    2.0

In [ ]:
data3.groupby('Attrited_Customer').agg({'Customer_Age':['mean','median'], 'Dependent_count':['mean','median']})

Customer_Age        Dependent_count       
                          mean median            mean median
Attrited_Customer                                           
0                    46.262118   46.0        2.335412    2.0
1                    46.659496   47.0        2.402581    2.0

Prueba de hipótesis

H0 = No hay diferencia entre las medias

HA = Si hay diferencias entre las medias

In [ ]:
mean_E_C = stats.ttest_ind(data3.groupby('Existing_Customer').agg({'Customer_Age':['mean']}), data3.groupby('Existing_Customer').agg({'Dependent_count':['mean']}))
mean_E_C #Es la media de Exiting

Ttest_indResult(statistic=array([218.80961477]), pvalue=array([2.08859191e-05]))

In [ ]:
if mean_E_C[1] > 0.05: #Es la media de Exiting
  print('se acepta la hipótesis nula, por lo que no existe diferencia entre las medias')
else:
  print('Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medias')

Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medias


**Conclusión intermedia:** a partir del p valor 2.08859191e-05<0.05 se rechaza la hipótesis nula, por lo que exite diferencia de medias entre Customer_Age y Dependent_Count para los datos existentes dentro de Existing_Customer

In [ ]:
mean_A_C = stats.ttest_ind(data3.groupby('Attrited_Customer').agg({'Customer_Age':['mean']}), data3.groupby('Attrited_Customer').agg({'Dependent_count':['mean']}))
mean_A_C #Es la media de Attrited

Ttest_indResult(statistic=array([218.80961477]), pvalue=array([2.08859191e-05]))

In [ ]:
if mean_A_C[1] > 0.05: #Es la media de Attrited
  print('se acepta la hipótesis nula, por lo que no existe diferencia entre las medias')
else:
  print('Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medias')

Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medias


**Conclusión intermedia:** a partir del p valor 2.08859191e-05<0.05 se rechaza la hipótesis nula, por lo que exite diferencia de medias entre Customer_Age y Dependent_Count para los datos existentes dentro de Attrited_Customer

In [ ]:
median_E_C = stats.ttest_ind(data3.groupby('Existing_Customer').agg({'Customer_Age':['median']}), data3.groupby('Existing_Customer').agg({'Dependent_count':['median']}))
median_E_C #Es la mediana de Exiting

Ttest_indResult(statistic=array([89.]), pvalue=array([0.00012622]))

In [ ]:
if median_E_C[1] > 0.05:#Es la mediana de Exiting
  print('se acepta la hipótesis nula, por lo que no existe diferencia entre las medianas')
else:
  print('Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medianas')

Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medianas


**Conclusión intermedia:** a partir del p valor 0.00012622<0.05 se rechaza la hipótesis nula, por lo que exite diferencia de medianas entre Customer_Age y Dependent_Count para los datos existentes dentro de Existing_Customer

In [ ]:
median_A_C = stats.ttest_ind(data3.groupby('Attrited_Customer').agg({'Customer_Age':['median']}), data3.groupby('Attrited_Customer').agg({'Dependent_count':['median']}))
median_A_C #Es la mediana de Attrited

Ttest_indResult(statistic=array([89.]), pvalue=array([0.00012622]))

In [ ]:
if median_A_C[1] > 0.05:#Es la mediana de Attrited
  print('se acepta la hipótesis nula, por lo que no existe diferencia entre las medianas')
else:
  print(f'Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medianas')

Se rechaza la hipótesis nula, por lo que si existe diferencia entre las medianas


**Conclusión intermedia:** a partir del p valor 0.00012622<0.05 se rechaza la hipótesis nula, por lo que exite diferencia de medianas entre Customer_Age y Dependent_Count para los datos existentes dentro de Attrited_Customer

#6 Construya cinco tablas de contingencia que relacionen por separado la variable Attrition Flag con:
##Gender##
##Education Level##
##Marital Status##
##Income Category##
##Card Category##
#Lleve a cabo, para cada tabla, una prueba chi2

In [ ]:
from scipy.stats import chi2_contingency

In [ ]:
pd.crosstab(data1['Gender'], data1['Attrition_Flag'], margins = True)

Attrition_Flag     0     1    All
Gender                           
0                697  4072   4769
1                930  4428   5358
All             1627  8500  10127

In [ ]:
st.chi2_contingency(pd.crosstab(data1['Gender'], data1['Attrition_Flag'], margins = True))[1]

0.007080482341208744

**Conclusión intermedia:** 

Prueba Chi2


H0: Si el p-valor>0.05, no hay diferencias significativas:

Ha: Si el p-valor<0.05, hay diferencias significativas

El p-valor que se obtiene de la prueba chi2 para combrobar la relación entre las variables Gender y Attrition_Flag es 0.007080482341208749<0.05, por consiguiente se rechaza la hipótesis nula por lo que podemos afirmar que hay diferencias significativas entre las variables Gender y Attrition_Flag



In [ ]:
data_3 = data[['Attrition_Flag', 'Education_Level']]
pd.crosstab(data_3['Attrition_Flag'], data_3['Education_Level'], margins = True)

Education_Level    College  Doctorate  Graduate  High School  Post-Graduate  \
Attrition_Flag                                                                
Attrited Customer      154         95       487          306             92   
Existing Customer      859        356      2641         1707            424   
All                   1013        451      3128         2013            516   

Education_Level    Uneducated  Unknown    All  
Attrition_Flag                                 
Attrited Customer         237      256   1627  
Existing Customer        1250     1263   8500  
All                      1487     1519  10127

In [ ]:
st.chi2_contingency(pd.crosstab(data_3['Attrition_Flag'], data_3['Education_Level']))[1]

0.05148913147336627

**Conclusión intermedia:** 

Prueba Chi2


H0: Si el p-valor>0.05, no hay diferencias significativas:

Ha: Si el p-valor<0.05, hay diferencias significativas

El p-valor que se obtiene de la prueba chi2 para comprobar la relación entre las variables Education_Level y Attrition_Flag es 0.5653199177295577>0.05, por consiguiente se acepta la hipótesis nula por lo que se cree que no hay diferencias significativas entre Education_Level y Attrition_Flag 

In [ ]:
data_4 = data[['Attrition_Flag', 'Marital_Status']]
pd.crosstab(data_4['Attrition_Flag'], data_4['Marital_Status'], margins = True)

Marital_Status     Divorced  Married  Single  Unknown    All
Attrition_Flag                                              
Attrited Customer       121      709     668      129   1627
Existing Customer       627     3978    3275      620   8500
All                     748     4687    3943      749  10127

In [ ]:
st.chi2_contingency(pd.crosstab(data_4['Attrition_Flag'], data_4['Marital_Status']))[1]

0.10891263394840227

**Conclusión intermedia:** 

Prueba Chi2


H0: Si el p-valor>0.05, no hay diferencias significativas:

Ha: Si el p-valor<0.05, hay diferencias significativas

El p-valor que se obtiene de la prueba chi2 para comprobar la relación entre las variables Marital_Status y Attrition_Flag es 0.6409480810705734>0.05, por consiguiente se acepta la hipótesis nula por lo que podemos afirmar que no hay diferencias significativas entre Marital_Status y Attrition_Flag.

In [ ]:
data_5 = data[['Attrition_Flag', 'Income_Category']]
pd.crosstab(data_5['Attrition_Flag'], data_5['Income_Category'], margins = True)

Income_Category    $120K +  $40K - $60K  $60K - $80K  $80K - $120K  \
Attrition_Flag                                                       
Attrited Customer      126          271          189           242   
Existing Customer      601         1519         1213          1293   
All                    727         1790         1402          1535   

Income_Category    Less than $40K  Unknown    All  
Attrition_Flag                                     
Attrited Customer             612      187   1627  
Existing Customer            2949      925   8500  
All                          3561     1112  10127

In [ ]:
st.chi2_contingency(pd.crosstab(data_5['Attrition_Flag'], data_5['Income_Category']))[1]

0.025002425704390617

**Conclusión intermedia:** 

Prueba Chi2


H0: Si el p-valor>0.05, no hay diferencias significativas:

Ha: Si el p-valor<0.05, hay diferencias significativas

El p-valor que se obtiene de la prueba chi2 para comprobar la relación entre las variables Marital_Status y Attrition_Flag es 0.38134984835176133>0.05, por consiguiente se acepta la hipótesis nula por lo que podemos afirmar que no hay diferencias significativas entre Marital_Status y Attrition_Flag.

In [ ]:
data_6 = data[['Attrition_Flag',	'Card_Category']]
pd.crosstab(data_6['Attrition_Flag'], data_6['Card_Category'], margins = True)

Card_Category      Blue  Gold  Platinum  Silver    All
Attrition_Flag                                        
Attrited Customer  1519    21         5      82   1627
Existing Customer  7917    95        15     473   8500
All                9436   116        20     555  10127

In [ ]:
st.chi2_contingency(pd.crosstab(data_6['Attrition_Flag'], data_6['Card_Category']))[1]

0.5252382797994759

**Conclusión intermedia:** 

Prueba Chi2


H0: Si el p-valor>0.05, no hay diferencias significativas:

Ha: Si el p-valor<0.05, hay diferencias significativas

El p-valor que se obtiene de la prueba chi2 para comprobar la relación entre las variables Marital_Status y Attrition_Flag es 0.9729757940389844>0.05, por consiguiente se acepta la hipótesis nula por lo que podemos afirmar que no hay diferencias significativas entre Card_Category y Attrition_Flag.

$$\huge\Conclusiones: $$ 


Dentro del análisis predente en este proyecto, se encontró especialmente que las variables analizadas presentaban diferencias significativas en cuanto a aspectos como la media y la mediana, lo que nos haría inferir que existe gran variabilidad de los datos, razón por la cual podemos pensar que la muestra de datos tuvo una correcta extracción. No obstante, al llevar a cabo la diferencia significativa de los datos entre variables mediante las tablas Crosstab, se encontró que en este caso no existían diferencias significattivas. En cuento al origen de distribución de las variables Customer_Age y Dependent_Count se encontró que estas no provenían de una distribución normal